web/src/activity.ts

```diff
@@ -58,8 +59,19 @@ export let client_is_active = document.hasFocus();
 // server-initiated reload as user activity.
 export let new_user_input = true;
 
+let received_new_messages = false;
+
+export function set_received_new_messages(value: boolean): void {
+    received_new_messages = value;
+}
+
 export function set_new_user_input(value: boolean): void {
     new_user_input = value;
+
+    if (received_new_messages && new_user_input) {
+        unread_ops.process_visible();
```

Comment:
I think a better way to handle this import cycle would be for `ui_init.ts` to pass an `on_new_user_input` handler that calls `unread_ops.process_visible` into `activity.initialize`.

But probably the nicest way to do it would be to have this module have a `on_new_user_input_hooks`, a la `compose_clear_box_hooks`.

Reply:
Added `on_new_user_input_hooks` to remove the dependecy cycle. Thanks!